File: OmarPakker/Source/QL_GOLD_C-Sharp/QL_Grammar/AST/Types/UnknownType.cs
using QL_Grammar.AST.Value;

namespace QL_Grammar.AST.Types
{
    public class UnknownType : IType
    {
        public static readonly UnknownType Instance = new UnknownType();

        public IValue DefaultValue { get { return new UnknownValue(); } }

        private UnknownType()
        {

        }

        public bool CompatibleWith(IType other)
        {
            return false;
        }

        public override string ToString()
        {
            return "unknown";
        }
    }
}

File: eenass/Myproject/src/ast/types/UnknownType.java
package ast.types;

import ast.Visitor;

public class UnknownType extends Type{

	public UnknownType() {
	}

	@Override
	public <T> T accept(Visitor<T> visitor) {
		return visitor.visit(this);
	}

	@Override
	public boolean isCompatibleTo(Type type) {
		return false;
	}
	
	public String toString(){
		return "Unknown";
	}

}
